File: spring/src/main/java/com/example/eurasia/entity/Data/CategorySelections.java
package com.example.eurasia.entity.Data;

public class CategorySelections implements Cloneable {

    private String category;
    private String[] selections;

    public CategorySelections() {
    }

    public CategorySelections(String category, String[] selections) {
        this.category = category;
        this.selections = selections.clone();//浅拷贝
    }

    public void setCategory(String category) {
        this.category = category;
    }
    public String getCategory(){
        return this.category;
    }
    public void setSelections(String[] selections) {
        this.selections = selections.clone();//浅拷贝
    }
    public String[] getSelections(){
        return this.selections;
    }

    @Override
    protected Object clone() throws CloneNotSupportedException {
        return super.clone();
    }
}

File: angular/src/app/user-conf/entities/category-selections.ts
export class CategorySelections {
  private catetory: string = null;
  private selections: string[] = null;
  public setCatetory(catetory: string) {
    this.catetory = catetory;
  }
  public getCategory(): string {
    return this.catetory;
  }
  public setSelections(selections: string[]): void {
    this.selections = this.selections;
  }
  public getSelections(): string[] {
    return this.selections;
  }
  public pushSelection(selection: string): void {
    this.selections.push(selection);
  }
  public constructor(catetory: string) {
    this.catetory = catetory;
    this.selections = [];
  }
}
